managed/resource_role.go

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright © 2022-present Yugabyte, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package managed
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	openapiclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+type resourceRoleType struct{}
+
+func (r resourceRoleType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `The resource to create a custom role in YugabyteDB Managed.`,
+		Attributes: map[string]tfsdk.Attribute{
+			"account_id": {
+				Description: "The ID of the account this role belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this role belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"role_id": {
+				Description: "The ID of the role. Created automatically when a role is created. Use this ID to get a specific role.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"role_name": {
+				Description: "The name of the role.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"role_description": {
+				Description: "The description of the role.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"permission_list": {
+				Description: "The permission list of the role.",
+				Required:    true,
+				Attributes: tfsdk.SetNestedAttributes(map[string]tfsdk.Attribute{
+					"resource_type": {
+						Type:     types.StringType,
+						Required: true,
+					},
+					"operation_groups": {
+						Type: types.SetType{
+							ElemType: types.StringType,
+						},
+						Required: true,
+					},
+				}),
+			},
+			"effective_permission_list": {
```

Comment:
What is the difference between the permission list and the effective permission list? Why are they not the same?

Reply:
Permission list is what the user has explicitly provided. But some permissions need to be given some more permissions as prerequisites to function correctly. We do that implicitly on the backend. The effective permission list is the final set of permissions given to the role including the implicit prerequisites

Reply:
Sorry, not trying to be obtuse here. What happens if the backend adds more implicit permissions? Will `terraform apply` break? Also, what if you do `terraform apply` twice and the list of permissions is different from the list of effective permissions? What if you make an implicit permission explicit - does `terraform apply` work in that case?

Reply:
"effective_permission_list" is always "known after apply" so I assumed that any changes during terraform apply would be fine.. is that wrong?

Reply:
> What if you make an implicit permission explicit - does terraform apply work in that case?

I did not find issues during this in my local testing but i will try out such cases again

Reply:
That should be how it works, but I have seen `terraform` do some interesting things in the past... So, I would like to ask you to do some manual testing of such scenarios ...  I will approve the PR, but please make sure to test such scenarios before merging.

Reply:
Definitely, thanks